app/front-end/src/features/editor/components/fileTreeView/fileTreeItem/fileTreeItemContextMenuTextfieldDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import { FileTreeViewItemProps } from '@/features/editor/types';
+import { FileTypes } from '@/types';
+import { Close as CloseIcon } from '@mui/icons-material';
+import {
+  Button,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Grid,
+  IconButton,
+  TextField,
+  Typography,
+  useTheme,
+} from '@mui/material';
+import { useEffect, useState } from 'react';
+
+export interface FileTreeItemContextMenuTextfieldDialogProps {
+  open: boolean;
+  title: string;
+  label: string;
+  item?: FileTreeViewItemProps;
+  onClose: () => void;
+  onSave: (label: string) => void;
+}
+
+/**
+ * `FileTreeItemContextMenuTextfieldDialog` component provides a dialog for editing the label of a file tree item.
+ *
+ * @description This component displays a modal dialog for users to input or edit a label for a file or folder in the file tree.
+ * It includes validation to ensure the input is not empty and does not exceed 50 characters. The dialog features a title
+ * and a text field for input, with a save and cancel button. The dialog is styled with the application's theme for a consistent
+ * look and feel.
+ *
+ * @component
+ *
+ * @param {FileTreeItemContextMenuTextfieldDialogProps} props - The props for the component.
+ * @param {boolean} props.open - A boolean indicating whether the dialog is open or closed.
+ * @param {string} props.title - The title of the dialog.
+ * @param {string} props.label - The label for the text field.
+ * @param {FileTreeViewItemProps} [props.item] - Optional file tree item object that includes the current label and file type.
+ * @param {() => void} props.onClose - Callback function to be called when the dialog is closed.
+ * @param {(label: string) => void} props.onSave - Callback function to be called when the user saves the input.
+ *
+ * @example
+ * // Example usage of the FileTreeItemContextMenuTextfieldDialog component
+ * <FileTreeItemContextMenuTextfieldDialog
+ *   open={dialogOpen}
+ *   title="Edit Label"
+ *   label="New Label"
+ *   item={fileTreeItem}
+ *   onClose={() => setDialogOpen(false)}
+ *   onSave={(newLabel) => console.log('New label:', newLabel)}
+ * />
+ *
+ * @returns {JSX.Element} The dialog component with a title, text field, and action buttons for saving or closing.
+ */
+export const FileTreeItemContextMenuTextfieldDialog: React.FC<FileTreeItemContextMenuTextfieldDialogProps> = ({
+  open,
+  title,
+  label,
+  item,
+  onClose,
+  onSave,
+}) => {
+  const Theme = useTheme();
+  const [value, setValue] = useState(item?.label || '');
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (open) {
+      setValue(item?.label || '');
+      setError(null);
+    }
+  }, [open, item?.label]);
+
+  const validateInput = (input: string) => {
+    if (!input.trim()) {
+      return 'Input cannot be empty';
+    }
+
+    if (input.length > 50) {
+      return 'Input must be less than 50 characters';
+    }
+
```

Comment:
We also need to validate if a file with specified name already exists

Reply:
Updated `workspaceContextProvider.tsx` to store file hierarchy and it can be accessed anywhere. New function in `utils/helpers.ts` provides ability to search for specific file. Added check for existing file while renaming or creating new file. Also added functionality to specify files' extension and removed the ability to define your own extension.

Reply:
Great work on the extensions! I just found one more small problem, it doesn't check for cases, I was able to create "A.csv" and "a.csv", it should be able to be solved by temporarily converting file names to lower-cases before checking if they exist.

Reply:
Updated `helpers.ts: doesFileExist` function